pipelines/build/regeneration/build_pipeline_generator.groovy

```diff
@@ -1,77 +1,283 @@
-node('master') {
-  def retiredVersions = [9, 10, 12, 13, 14, 15]
+import java.nio.file.NoSuchFileException
+import groovy.json.JsonSlurper
+import groovy.json.JsonOutput
 
-  (8..30).each({javaVersion -> 
+node('master') {
+  try {
```

Comment:
If any of this can be split into functions that would be ace.

Reply:
This looks ok to leave as is for now. Functionising it will probably make it overly complex and not benefit us much